cmd/kubeadm/app/util/config/cluster.go

```diff
@@ -136,9 +141,27 @@ func GetNodeRegistration(kubeconfigFile string, client clientset.Interface, node
 		return errors.Wrap(err, "failed to get corresponding node")
 	}
 
-	criSocket, ok := node.ObjectMeta.Annotations[constants.AnnotationKubeadmCRISocket]
-	if !ok {
-		return errors.Errorf("node %s doesn't have %s annotation", nodeName, constants.AnnotationKubeadmCRISocket)
+	var criSocket string
+	var ok bool
+	if features.Enabled(clusterCfg.FeatureGates, features.NodeLocalCRISocket) {
```

Comment:
So, to add hostname support too, it would be something similar to this code block here, up at:
> nodeName, err := getNodeNameFromKubeletConfig(kubeconfigFile)
?

Reply:
not sure what hostname support means, but instance-config.yaml can have any kubeletconfiguration field.

Reply:
I don't quite understand what you mean, but the kubelet configuration does not contain a nodeName field.

Reply:
ah. sorry. yes, I mean node name, not host name.